agent/taskresource/volume/volume.go

```diff
@@ -25,25 +27,29 @@ type VolumeResource struct {
 	// Mountpoint is a read-only field returned from docker
 	Mountpoint          string
 	Driver              string
+	DriverOpts          map[string]string
 	Labels              map[string]string
 	createdAtUnsafe     time.Time
 	desiredStatusUnsafe VolumeStatus
 	knownStatusUnsafe   VolumeStatus
+	client              engine.DockerClient
 	// lock is used for fields that are accessed and updated concurrently
 	lock sync.RWMutex
 }
 
 // NewVolumeResource returns a docker volume wrapper object
-func NewVolumeResource(name string, 
-	mountPoint string, 
-	driver string, 
-	labels map[string]string) *VolumeResource {
+func NewVolumeResource(name string,
```

Comment:
The `VolumeResource` seems can be private?

Reply:
this PR lacks code that actually uses this yet. Once we replace empty volume container, it'll have to be referenced. If not, I'll update it to be private then.